projects/client-ws/connector.ts

```diff
@@ -0,0 +1,66 @@
+import { ClientConnector, connectClient, applyApi, Client } from '@remixproject/plugin/connector'
+import { PluginClient } from '@remixproject/plugin/client'
+import { Message } from '../utils/src/types/message'
+import { RemixApi } from '../utils/src/api/remix-profile'
+import { Api, ApiMap } from '../utils/src/types/api'
+
+
+export interface WS {
+  send(data: string): void
+  on(type: 'message', cb: (event: string) => any): this
+}
+
+/**
+ * This Websocket connector works with the library `ws`
+ */
+export class WebsocketConnector implements ClientConnector {
+
+  constructor(private websocket: WS) {}
+
+  /** Send a message to the engine */
+  send(message: Partial<Message>) {
+    this.websocket.send(JSON.stringify(message))
```

Comment:
should be `websocket.send(JSON.stringify(message))` ?

Reply:
No, this is the local websocket pass as parameter in the constructor. Using `private` in the parameter of a constructor is a shortcut for making it a property of the class.
I use that so I'm not dependant on the websocket implementation (it's based on the "ws" library, but could be used by another).